doc/api/errors.markdown

```diff
@@ -0,0 +1,448 @@
+//     at Function.Module.runMain (module.js:497:10)
+//     at startup (node.js:119:16)
+//     at node.js:906:3
+```
+
+The location information will be one of: 
+
+* `native`, if the frame represents a call internal to V8 (as in `[].forEach`).
+* `plain-filename.js:line:column`, if the frame represents a call internal to Node.
+* `/absolute/path/to/file.js:line:column`, if the frame represents a call in a user program, or its dependencies.
+
+It is important to note that the string representing the stacktrace is only
+generated on **access**: it is lazily generated. 
+
+The number of frames captured by the stack trace is bounded by the smaller of
+`Error.stackTraceLimit` or the number of available frames on the current event
+loop tick.
+
+System-level errors are generated as augmented Error instances, which are detailed
+[below](#errors_system_errors).
+
+#### Error.captureStackTrace(targetObject)
+
+Creates a `.stack` property on `targetObject`, which when accessed returns
+a string representing the location in the program at which `Error.captureStackTrace`
+was called.
+
+```javascript
+var myObject = {};
+
+Error.captureStackTrace(myObject);
+
+myObject.stack  // similar to `new Error().stack`
+```
+
+The first line of the trace, instead of being prefixed with `ErrorType:
+message`, will be the result of `targetObject.toString()`.
+
+#### Error.stackTraceLimit
+
+Property that determines the number of stack frames collected by a stack trace
+(whether generated by `new Error().stack` or `Error.captureStackTrace(obj)`).
+
+The initial value is `10`. It may be set to any valid JavaScript number, which
+will effect any stack trace captured *after* the value has been changed. If set
+to a non-number value, stack traces will not capture any frames and will report
+`undefined` on access.
+
+### Class: RangeError
+
+A subclass of Error that indicates that a provided argument was not within the
+set or range of acceptable values for a function; whether that be a numeric
+range, or outside the set of options for a given function parameter. An example:
+
+```javascript
+require('net').connect(-1);  // throws RangeError, port should be > 0 && < 65536
+```
+
+Node will generate and throw RangeError instances *immediately* -- they are a form
+of argument validation. 
+
+### Class: TypeError
+
+A subclass of Error that indicates that a provided argument is not an allowable
+type. For example, passing a function to a parameter which expects a string would
+be considered a TypeError.
+
+```javascript
+require('url').parse(function() { }); // throws TypeError, since it expected a string
+```
+
+Node will generate and throw TypeError instances *immediately* -- they are a form
+of argument validation.
+
+### Class: ReferenceError
+
+A subclass of Error that indicates that an attempt is being made to access a variable
+that is not defined. Most commonly it indicates a typo, or an otherwise broken program.
+Only V8 may generate and propagate these errors.
```

Comment:
Not technically true. It's possible to do a `new ReferenceError('foo');`, but there's no reason to.